samza-core/src/test/scala/org/apache/samza/config/TestStorageConfig.scala

```diff
@@ -29,12 +29,26 @@ class TestStorageConfig {
   @Test
   def testIsChangelogSystem {
     val configMap = Map[String, String](
-      FACTORY.format("system1") -> "some.factory.Class",
-      CHANGELOG_STREAM.format("system1") -> "system1.stream1",
-      FACTORY.format("system2") -> "some.factory.Class")
+      FACTORY.format("store1") -> "some.factory.Class",
+      CHANGELOG_STREAM.format("store1") -> "system1.stream1",
+      FACTORY.format("store2") -> "some.factory.Class")
     val config = new MapConfig(configMap)
     assertFalse(config.isChangelogSystem("system3"))
     assertFalse(config.isChangelogSystem("system2"))
     assertTrue(config.isChangelogSystem("system1"))
   }
+
+  @Test
+  def testIsChangelogSystemSetting {
+    val configMap = Map[String, String](
+      FACTORY.format("store1") -> "some.factory.Class",
+      CHANGELOG_STREAM.format("store1") -> "system1.stream1",
```

Comment:
Also, with this change, I think that we will need to update our internal config rewriter as well, to detect all systems used by the job.